fluent/reflect.go

```diff
@@ -0,0 +1,207 @@
+package fluent
+
+import (
+	"fmt"
+	"reflect"
+	"sort"
+
+	"github.com/ipld/go-ipld-prime"
+)
+
+func Reflect(np ipld.NodePrototype, i interface{}) (ipld.Node, error) {
+	return defaultReflector.Reflect(np, i)
+}
+
+func ReflectIntoAssembler(na ipld.NodeAssembler, i interface{}) error {
+	return defaultReflector.ReflectIntoAssembler(na, i)
+}
+
+var defaultReflector = Reflector{
+	MapOrder: func(x, y string) bool {
+		return x < y
+	},
+}
+
+type Reflector struct {
+	// MapOrder is used to decide a deterministic order for inserting entries to maps.
+	// (This is used when converting golang maps, since their iteration order is randomized;
+	// it is not used when converting other types such as structs, since those have a stable order.)
+	// MapOrder should return x < y in the same way as sort.Interface.Less.
+	MapOrder func(x, y string) bool
```

Comment:
why make this configurable? you need quite a bit more code for it, and I can't imagine why anyone would want a different order.

Reply:
I regret I feel obliged to mention that the CBOR RFC specifies a sort order that is Not What You'd (Or Anyone'd) Expect.

Reply:
We're in Data Model land up here, so technically it could be considered Not This Layer's Problem, but I'm of the thinking that if someone wants to have an application that uses some canonical sort or other at the codec level, it's optimal if we can get things in the desired order as soon as possible and thereafter just keep them that way.

Reply:
Yeah, agreed, I guess we can just call this "unfortunate" :) Marking this as resolved.